speechbrain/utils/hparams.py

```diff
@@ -20,6 +20,10 @@ def choice(value, choices, default=None):
         to the corresponding return values
     default: any
         the default value
+    apply: bool
+        if set to true, the value is expected to
```

Comment:
Actually, in this case it is not an index... The way choice is used is as follows.

```yaml
spk_emb_discrete_src: !apply:speechbrain.utils.hparams.choice
    value: !ref <ssl_model_type>
    choices:
        wavlm: flexthink/discrete_wavlm_spk_rec_ecapatdn_lite
        hubert: flexthink/discrete_hubert_spk_rec_ecapatdn_lite
        wav2vec2: flexthink/discrete_wav2vec2_spk_rec_ecapatdn_lite
```

So the `value` indicates the value to be mapped using choices... Perhaps it could be renamed to `key` by analogy to dictionary keys but it is not a numeric index.

The most typical use case for this is this:

* You have an hparam indicating, for instance, the type of base model to use
* Multiple other params will be automatically selected based on this choice

Without `choice` you'd have to create different hparams files for all the permutations and combinations.